src/main/java/io/rubuy74/rhs/adapter/in/web/EventService.java

```diff
@@ -0,0 +1,44 @@
+package io.rubuy74.rhs.adapter.in.web;
+
+import io.rubuy74.rhs.domain.Event;
+import io.rubuy74.rhs.exception.EventListingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.retry.annotation.Backoff;
+import org.springframework.retry.annotation.Retryable;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClient;
+
+import java.util.List;
+
+@Service
+public class EventService {
+    public static Logger logger = LoggerFactory.getLogger(EventService.class);
+
+    @Retryable(maxAttempts = 3, backoff = @Backoff(delay = 1000))
+    public List<Event> getEvents(RestClient restClient) {
+        ParameterizedTypeReference<List<Event>> responseType =
+                new ParameterizedTypeReference<>() {};
+
+        return restClient.get()
+                .uri("/events")
+                .retrieve()
+                .onStatus(HttpStatusCode::is4xxClientError,
+                        (request,response)-> {
+
+                            logger.error("Failed to retrieve events. Status: {}", response.getStatusCode());
```

Comment:
usually key/value pairs are better optimised for log searching take a look at the [logfmt](https://brandur.org/logfmt) this is one of the formats that we commonly use

```suggestion
                            logger.error("operation=getEvents, msg='Failed to retrieve events.', status='{}'", response.getStatusCode());
```

Reply:
Thanks for the tip.

I'll use this for all my logs from now on. 😄